### PR TITLE
fix(components/ag-grid): avoid unneeded api calls for row selection

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2474974](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2474974)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1135/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/a11y-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/ag-grid-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/core-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/indicators-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/pages-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/popovers-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/tabs-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1135/storybooks/tiles-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1135/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1135/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1135/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1135/playground/)

**Review by Blackbaud-SteveBrush — DISMISSED** [review]
Again, great-looking code!

File: libs/components/ag-grid/src/lib/modules/ag-grid/cell-renderers/cell-renderer-row-selector/cell-renderer-row-selector.component.spec.ts [hunk 1]
```diff
@@ -174,11 +174,40 @@ describe('SkyCellRendererCheckboxComponent', () => {
...
+      expect(
+        rowSelectorCellComponent.rowNode?.setSelected
+      ).not.toHaveBeenCalled();
+    });
   });
 
   describe('refresh', () => {
     it('returns false', () => {
-      expect(rowSelectorCellComponent.refresh()).toBe(false);
+      expect(
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/ag-grid/src/lib/modules/ag-grid/cell-renderers/cell-renderer-row-selector/cell-renderer-row-selector.component.spec.ts:204]: I think this `it` description needs to be updated

**Review by Blackbaud-ErikaMcVey — COMMENTED** [review]
One comment, otherwise looks good. Did you test this with data manager select/deselect all? Just want to make sure external selection still works

File: libs/components/ag-grid/src/lib/modules/ag-grid/cell-renderers/cell-renderer-row-selector/cell-renderer-row-selector.component.spec.ts (hunk 1 above)
Comment (johnhwhite) [reply libs/components/ag-grid/src/lib/modules/ag-grid/cell-renderers/cell-renderer-row-selector/cell-renderer-row-selector.component.spec.ts:204]: Good catch. Updated.

**Review by Blackbaud-ErikaMcVey — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See null

**Pull request merged**